rotkehlchen/chain/evm/decoding/curve/decoder.py

```diff
@@ -547,6 +570,71 @@ def _decode_curve_trades(self, context: DecoderContext) -> DecodingOutput:
 
         return DEFAULT_DECODING_OUTPUT
 
+    def _pre_decode_deposit_and_stake(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Enrich the transfer for deposit and stake to save the amount of gauge tokens
+        received. We need to match against transfers manually because they are not
+        made from the user address but an intermediate contract
+        """
+        if (pool_assets := self.pools.get(context.tx_log.address)) is None:
+            log.error(
+                f'Curve pool for {context.tx_log.address} not present in cache at '
+                f'{context.transaction.tx_hash.hex()}. Skipping',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        provider = hex_or_bytes_to_address(context.tx_log.topics[1])
+        deposited_amounts = [
+            hex_or_bytes_to_int(deposit_amount)
+            for deposit_amount in [context.tx_log.data[i:i + 32] for i in range(len(pool_assets))]
+        ]
+        resolved_assets = [
```

Comment:
These are not resolved assets. I don't see any resolution, just the asset constructor. In fact you are resolving them only in special cases below. So the name here is dangerously misleading. Please use better names.

These should be called `pool_assets`. And what you iterate is not assets, but addresses. Again ... naming.

```suggestion
        pool_assets = [
```